### PR TITLE
hwmv2: Port STM32WLX SoCs & boards

### PR DESCRIPTION
**Pull request by ajarmouni-st**
Port STM32WLX SoCs & boards to Hardware Model V2.

### CODE REVIEW

File: soc/st/stm32/stm32wl/Kconfig.soc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+config SOC_STM32WL55XX
+	bool
+	select SOC_SERIES_STM32WLX
+
+config SOC
+	default "STM32WLE4XX" if SOC_STM32WLE4XX
+	default "STM32WLE5XX" if SOC_STM32WLE5XX
+	default "STM32WL54XX" if SOC_STM32WL54XX
+	default "STM32WL55XX" if SOC_STM32WL55XX
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply soc/st/stm32/stm32wl/Kconfig.soc:33]: Yes, user lower case here to match the soc.yml file

File: boards/olimex/olimex_lora_stm32wl_devkit/board.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: olimex_lora_stm32wl_devkit
+  vendor: olimex
+  socs:
+    - name: stm32wle5xx
```
Comment (erwango) [inline boards/olimex/olimex_lora_stm32wl_devkit/board.yml:5]: Missing revisions

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ajarmouni-st** [comment]
Rebased to resolve conflicts & address comments above.

**Comment by nordicjm** [comment]
rebase needed

File: boards/olimex/olimex_lora_stm32wl_devkit/revision.cmake
Comment (erwango) [inline boards/olimex/olimex_lora_stm32wl_devkit/revision.cmake:1]: This file could be removed.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/olimex/olimex_lora_stm32wl_devkit/revision.cmake
Comment (ajarmouni-st) [reply boards/olimex/olimex_lora_stm32wl_devkit/revision.cmake:1]: In stm32f411e_disco too.

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
Please rebase

**Comment by ajarmouni-st** [comment]
> Please rebase

Done

**Review by nordicjm — COMMENTED** [review]
Haven't checked but does documentation/links need updating e.g. https://github.com/zephyrproject-rtos/zephyr/pull/68801 ?

**Comment by ajarmouni-st** [comment]
> ajarmouni-st requested review from yonsch, fkokosinski, kgugala, stephanosio, ibirnbaum, carlescufi, MulinChao, ChiHuaL, Mani-Sadhasivam, mmahadevan108, dleach02, rerickson1, simonguinot, parthitce, jackrosenthal, nandojve, ifyall, npal-cy, vanti, greg-leach, LingaoM, soburi and avisconti as code owners 1 hour ago

Sorry everyone for the inconvenience, it was done automatically by zephyrbot.

**Comment by ajarmouni-st** [comment]
> Haven't checked but does documentation/links need updating e.g. #68801 ?

done

**Review by erwango — APPROVED** [review]

File: boards/seeed/lora_e5_dev_board/board.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: lora_e5_dev_board
+  vendor: seeed
```
[the remark this answers is not in the document]
Comment (ajarmouni-st) [reply boards/seeed/lora_e5_dev_board/board.yml]: fixed

File: boards/seeed/lora_e5_mini/doc/index.rst [hunk 4]
```diff
@@ -100,10 +100,10 @@ The Zephyr LoRa-E5 mini configuration supports the following hardware features:
 
 Other hardware features are not yet supported on this Zephyr port.
 
-The default configuration can be found in the defconfig and dts files:
+The default configuration can be found in:
 
-- :zephyr_file:`boards/arm/lora_e5_mini/lora_e5_mini_defconfig`
-- :zephyr_file:`boards/arm/lora_e5_mini/lora_e5_mini.dts`
+- :zephyr_file:`boards/seed/lora_e5_mini/lora_e5_mini_defconfig`
+- :zephyr_file:`boards/seed/lora_e5_mini/lora_e5_mini.dts`
```
[the remark this answers is not in the document]
Comment (ajarmouni-st) [reply boards/seeed/lora_e5_mini/doc/index.rst]: fixed

File: boards/seeed/lora_e5_dev_board/board.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: lora_e5_dev_board
+  vendor: seeed studio
```
Comment (nordicjm) [inline boards/seeed/lora_e5_dev_board/board.yml:3]:
```suggestion
  vendor: Seeed Studio
```

File: boards/seeed/lora_e5_dev_board/board.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+board:
```
Comment (nordicjm) [inline boards/seeed/lora_e5_dev_board/board.yml:1]: also move from folder `seeed` -> `seeed_studio`

File: boards/seeed/lora_e5_mini/board.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: lora_e5_mini
+  vendor: seeed studio
```
Comment (nordicjm) [inline boards/seeed/lora_e5_mini/board.yml:3]: as above

File: boards/seeed/lora_e5_mini/doc/index.rst [hunk 8]
```diff
@@ -100,10 +100,10 @@ The Zephyr LoRa-E5 mini configuration supports the following hardware features:
 
 Other hardware features are not yet supported on this Zephyr port.
 
-The default configuration can be found in the defconfig and dts files:
+The default configuration can be found in:
 
-- :zephyr_file:`boards/arm/lora_e5_mini/lora_e5_mini_defconfig`
-- :zephyr_file:`boards/arm/lora_e5_mini/lora_e5_mini.dts`
+- :zephyr_file:`boards/seeed/lora_e5_mini/lora_e5_mini_defconfig`
+- :zephyr_file:`boards/seeed/lora_e5_mini/lora_e5_mini.dts`
```
Comment (nordicjm) [inline boards/seeed/lora_e5_mini/doc/index.rst:106]: and update filenames here and in other file

File: boards/seeed/lora_e5_dev_board/board.yml (hunk 5 above)
Comment (ajarmouni-st) [reply boards/seeed/lora_e5_dev_board/board.yml]: Is this required? I noticed on other boards it's not enforced,

File: boards/seeed/lora_e5_dev_board/board.yml (hunk 6 above)
Comment (ajarmouni-st) [reply boards/seeed/lora_e5_dev_board/board.yml]: fixed

File: boards/seeed/lora_e5_mini/doc/index.rst (hunk 8 above)
Comment (ajarmouni-st) [reply boards/seeed/lora_e5_mini/doc/index.rst]: fixed

**Review by nordicjm — APPROVED** [review]

**Pull request merged**